### PR TITLE
Add FastSettlementMiddleware

### PR DESCRIPTION
**Pull request by zkJoaquin**
fix: #113

### CODE REVIEW

File: contracts/fast-settlement/FastSettlementMiddleware.sol [hunk 1]
```diff
@@ -1,65 +1,77 @@
...
+
+    EnumerableSetUpgradeable.AddressSet private operators;
+    EnumerableSetUpgradeable.AddressSet private vaults;
+
+    // @notice The risk factor is used to calculate credit for all tokens if it's not set in tokenRiskFactor
+    uint256 public riskFactor;
+    // @notice The token risk factor will override the riskFactor if it's set
+    mapping(address => uint256) public tokenRiskFactor;
+
+    IArbitrator public arbitrator;
```
Comment (zkbenny) [inline contracts/fast-settlement/FastSettlementMiddleware.sol:38]: immutable

File: contracts/fast-settlement/FastSettlementMiddleware.sol [hunk 2]
```diff
@@ -131,4 +198,16 @@ contract FastSettlementMiddleware is Ownable, IFastSettlementMiddleware {
...
+    /// @notice Return the risk factor for token
+    /// @dev The risk factor will not be less than 1
+    function getTokenRiskFactor(address _token) public view returns (uint256) {
+        uint256 risk = tokenRiskFactor[_token];
+        return
+            risk > 0
+                ? risk
+                : riskFactor > 0
+                    ? riskFactor
+                    : 1;
```
Comment (zkbenny) [inline contracts/fast-settlement/FastSettlementMiddleware.sol:211]: default: RISK_FACTOR_DENUMINATOR

File: contracts/fast-settlement/FastSettlementMiddleware.sol (hunk 1 above)
Comment (zkJoaquin) [reply contracts/fast-settlement/FastSettlementMiddleware.sol]: fixed

File: contracts/fast-settlement/FastSettlementMiddleware.sol (hunk 2 above)
Comment (zkJoaquin) [reply contracts/fast-settlement/FastSettlementMiddleware.sol]: fixed

**Review by zkbenny — APPROVED** [review]

**Pull request merged**